### ISSUE TITLE
 A load balancer cannot be attached to multiple subnets in the same Availability Zone

### ISSUE THREAD

**Issue by koen-venly** [robertdebock/terraform-aws-vault#5]
Error: error creating application Load Balancer: InvalidConfigurationRequest: A load balancer cannot be attached to multiple subnets in the same Availability Zone
│       status code: 400, request id: 9831def4-be92-4315-a399-ddeb1b8f9dde
│ 
│   with module.vault.aws_lb.default,
│   on .terraform/modules/vault/main.tf line 270, in resource "aws_lb" "default":
│  270: resource "aws_lb" "default" {

**Comment by robertdebock** [comment]
Related code:

```hcl
# Add a load balancer.
resource "aws_lb" "default" {
  load_balancer_type = "application"
  name               = var.name
  security_groups    = [aws_security_group.loadbalancer.id, aws_security_group.default.id]
  subnets            = local.aws_subnet_ids
  tags               = var.tags
}
```

The related local:

```hcl
  # Set the `aws_subnet_ids` based on either the resource object or the data object, whichever is set.
  aws_subnet_ids = coalescelist(aws_subnet.default[*].id, try(tolist(data.aws_subnets.default[0].ids), []))
```

Related data object:

```hcl
data "aws_subnets" "default" {
  count  = var.vpc_id == "" ? 0 : 1
  filter {
    name   = "vpc-id"
    values = [local.vpc_id]
  }
}
```

So; I guess the data object (`aws_subnets`) needs an extra `filter`.

**Comment by robertdebock** [comment]
Just released version 2.0.0, which addresses this issue.
